spring-boot-sample-project/src/main/java/com/example/demo/controller/UserController.java

```diff
@@ -0,0 +1,53 @@
+package com.example.demo.controller;
+
+import com.example.demo.pojo.User;
+import com.example.demo.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/")
+public class UserController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/user")
+    public ResponseEntity<List<User>> getAllUser(){
+        List<User> userList = userService.getAllUser();
+        return ResponseEntity.ok(userList);
+    }
+
+    @GetMapping("/user/{name}")
+    public ResponseEntity<User> getUser(@PathVariable(value="name") String userName){
+        User user = userService.getAUser(userName);
+        return ResponseEntity.ok(user);
+    }
+
+    @PostMapping("/user")
+    public ResponseEntity<String> createUser(@Valid @RequestBody User user){
+        if(userService.createUser(user)){
+            return ResponseEntity.ok("User Created");
+        }
+        return ResponseEntity.ok("Cannot create a user");
```

Comment:
there could be multiple reasons why a user cannot be created. If it is due to bad request body, you should throw a `400`. If it can be due to some  reason which is handled by your application code, you should return
```
{
   "success": false,
  "errorCode": "ALREADY_SIGNED_UP"
}
```
or something like that.

In this case, when entity creation is successful, you should return `success: true` with `200` and not `201` as I mentioned earlier to be consistent